cirq/ops/common_channels.py

```diff
@@ -31,109 +31,192 @@
 class AsymmetricDepolarizingChannel(gate_features.SingleQubitGate):
     """A channel that depolarizes asymmetrically along different directions."""
 
-    def __init__(self, p_x: float, p_y: float, p_z: float) -> None:
+    def __init__(self,
+                 p_x: Optional[float] = None,
+                 p_y: Optional[float] = None,
+                 p_z: Optional[float] = None,
+                 error_probabilities: Optional[Dict[str, float]] = None
+                ) -> None:
         r"""The asymmetric depolarizing channel.
 
-        This channel applies one of four disjoint possibilities: nothing (the
-        identity channel) or one of the three pauli gates. The disjoint
-        probabilities of the three gates are p_x, p_y, and p_z and the
-        identity is done with probability 1 - p_x - p_y - p_z. The supplied
-        probabilities must be valid probabilities and the sum p_x + p_y + p_z
-        must be a valid probability or else this constructor will raise a
-        ValueError.
+        This channel applies one of 4**n disjoint possibilities: nothing (the
+        identity channel) or one of the 4**n - 1 pauli gates.
 
         This channel evolves a density matrix via
 
             $$
-            \rho \rightarrow (1 - p_x - p_y - p_z) \rho
-                    + p_x X \rho X + p_y Y \rho Y + p_z Z \rho Z
+            \sum_i p_i Pi \rho Pi
             $$
 
+        where i varies from 0 to 4**n-1 and Pi represents n-qubit Pauli operator
+        (including identity). The input \rho is the density matrix before the
+        depolarization.
+
         Args:
             p_x: The probability that a Pauli X and no other gate occurs.
             p_y: The probability that a Pauli Y and no other gate occurs.
             p_z: The probability that a Pauli Z and no other gate occurs.
+            error_probabilities: Dictionary of string (Pauli operator) to its
+                probability
+
+        Examples of calls:
+            * Single qubit: AsymmetricDepolarizingChannel(0.2, 0.1, 0.3)
+            * Single qubit: AsymmetricDepolarizingChannel(p_z=0.3)
+            * Two qubits: AsymmetricDepolarizingChannel(
+                                error_probabilities={'XX': 0.2})
 
         Raises:
             ValueError: if the args or the sum of args are not probabilities.
         """
+        if error_probabilities:
+            num_qubits = len(list(error_probabilities)[0])
+            for k in error_probabilities.keys():
+                if not set(k).issubset({'I', 'X', 'Y', 'Z'}):
+                    raise ValueError(f"{k} is not made solely of I, X, Y, Z.")
+                if len(k) != num_qubits:
+                    raise ValueError(f"{k} must have {num_qubits} Pauli gates.")
+            self._num_qubits = num_qubits
+            self._error_probabilities = error_probabilities
```

Comment:
Perform validation of each probability individually and the overall sum in a way that generalizes the validation below for the case of single-qubit.